modules/juce_core/native/juce_mac_SystemStats.mm

```diff
@@ -71,6 +71,13 @@ static void doCPUID (uint32& a, uint32& b, uint32& c, uint32& d, uint32 type)
 //==============================================================================
 void CPUInformation::initialise() noexcept
 {
+   #if __FMA__

```

Comment:
This define will be set if we are compiling with FMA/FMA4. This CPUInformation checks are about runtime. FMA is similar to SSE3 but bit 12 rather than 0. FMA4 is in a different page and will need a new  SystemStatsHelpers::doCPUID call.

Wikipedia has a good list but if you dont trust it double check with the Intel guide

Reply:
> This define will be set if we are compiling with FMA/FMA4. This CPUInformation checks are about runtime.

Hm, good point and makes sense!

> FMA is similar to SSE3 but bit 12 rather than 0. FMA4 is in a different page and will need a new SystemStatsHelpers::doCPUID call.

Ahh, thanks! For the most part, I took this task on to learn more about how this all works... this is helping!

Reply:
You should be able to call again with 0x80000001 then check bit 16 of c for fma4. It seems that extension is AMD only so almost no one will have that available (no AMD mac machines officially).

Reply:
Is there a risk of borking anything when running on iOS when performing such a call? I don't have a way to test that platform to be sure the system isn't throwing red-flags of any kind. I just know that such a highly sandboxed system tends to be sensitive.

Reply:
The block that checks them just below is inside a `#if JUCE_INTEL && ! JUCE_NO_INLINE_ASM` so it will not be done on iOS

Reply:
Nice, good to know

Reply:
@hjmallon Where did you find that detail, btw?